dynamic_programming/house_robber.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file
+ * @brief Implementation of [House Robber Problem] algorithm
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Reply:
Done 👍